sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1582 @@
+
+### Fixed: Expiration of `contract-caller` Allowances
+
+The `pox` contract's implementation of contract-caller allowance
+expirations is broken. `pox-2` should fix this behavior.
+
+### Usecase: Stacking without a Cooldown Cycle
+
+#### Direct stackers
+
+Direct stackers wishing to repeatedly stack without cooldown may
+issue a simple contract-call transaction, invoking:
+
+     `SP000000000000000000002Q6VF78.pox-2 stack-extend`
+
+If the user wishes to *increase* the amount that they have stacked
+in addition to repeating the stacking lockup, they can separately issue
+a contract-call transaction to:
+
+     `SP000000000000000000002Q6VF78.pox-2 stack-increase`
+
+Users can launch a utility smart-contract to execute both of these
+contract-calls atomically, but they must remember to invoke the
+contract caller allowance for that utility contract.
+
+#### Delegated stackers
+
+Delegation operators wishing to repeatedly stack with a given user
+without cooldown may do the following:
+
+1. Issue a contract-call on behalf of the delegated stacker they wish
+   to re-stack:
+   
+```clarity
+   (contract-call SP000000000000000000002Q6VF78.pox-2 delegate-stack-extend
+     ...)
+```
+
+2. That contract-call *locks but does not commit* the user's funds. In
+   order for the delegation operator to register the reward address for
+   those funds, they must invoke the aggregation commit function (just as
+   when they invoke `delegate-stack-stx`):
+
+```clarity
+   (contract-call SP000000000000000000002Q6VF78.pox-2 stack-aggregation-commit
+     ...)
+```
+
+If the delegator wishes to *increase* the amount that the user has
+stacked, they must separately issue a `delegate-stack-increase` call
+which locks but does not commit the increased funds.
+
+### Usecase: Stacking with Multiple PoX Addresses
+
+#### Direct stackers
+
+Direct stackers cannot use multiple reward address for the same Stacking
+address. This is because the direct stacking interface does not support
+partial stacking operations (i.e., separating the locking operation from
+the stacking operation), which are necessary for supporting this use case.
+Users who wish to do this should use the delegation interfaces.
+
+#### Delegated stackers
+
+Even in PoX-1, delegation operators are capable of assigning multiple
+reward addresses for funds that they manage: each invocation of
+`delegate-stack-stx` can use a different PoX reward address. However,
+if operators wished to split the same _account's_ lockups across
+multiple addresses, this would not be possible.
+
+In PoX-2, however, delegation operators can use
+`delegate-stack-increase` to achieve this. This method allows the
+delegation operator to set a new PoX address to receive the
+"partially stacked" funds from the increase. Once called, the operator
+can use `stack-aggregation-commit` to commit to each PoX reward
+address separately.
+
+## Clarity
```

Comment:
This sections begins with 'new methods' that will be introduced in Clarity 2, then has two 'Changed' & 'Fixed' sub-sections at the end. Suggest flipping that order so that the the 'changed' and 'fixed' are first then followed by the newly added methods and features. 

Additionally adding a hyperlink and reference at the bottom to the point curious readers to the documentation

I leave it your best judgement if that should be clarity-lang.org or docs.stacks.co link

Reply:
The reason the order is "new", "changed", and then "fixed" is because it's consistent with how most changelogs are written (which is [advised](https://github.com/stacksgov/sips/blob/feat/sip-015/sips/sip-015/sip-015-network-upgrade.md#how-to-read-this-sip) at the start of the SIP).  See the [changelog spec](https://keepachangelog.com/en/1.0.0/) for details.